### PR TITLE
Don't run terraform operations in case the AWS credentials are about to expire

### PR DESCRIPTION
**Pull request by mujx**
## Description

The cli tool will check if the AWS credentials are going to expire in the next 5 minutes and stop the execution.

## Motivation and Context

https://github.com/terraform-providers/terraform-provider-aws/issues/4502

## How Has This Been Tested?

Using AWS credentials valid for 1h and setting the `AWS_PROFILE` env variable.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/ops/terraform/terraform_cmd_generator.py [hunk 1]
```diff
@@ -27,6 +31,11 @@ def __init__(self, root_dir, cluster_config, inventory_generator, ops_config, te
         self.template = template
 
     def generate(self, args):
+        if 'AWS_PROFILE' in os.environ:
```
Comment (amuraru) [inline src/ops/terraform/terraform_cmd_generator.py]: Is this env var set by opscli before running tf?

Comment (mujx) [reply src/ops/terraform/terraform_cmd_generator.py]: This is one of the default variables used by the aws cli. https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

File: src/ops/cli/aws.py [hunk 2]
```diff
@@ -9,9 +9,19 @@
...
 def acess_key(profile):
     return get_output('aws configure get aws_access_key_id --profile %s' % profile)
 
 def secret_key(profile):
     return get_output('aws configure get aws_secret_access_key --profile %s' % profile)
+
+def expiry_dttm(profile):
+    return get_output('aws configure get expiry_dttm --profile %s' % profile)
+
+def expiry_dttm_in_minutes(profile):
```
Comment (costimuraru) [inline src/ops/cli/aws.py]: Not familiar with the `dttm` acronym, what does it stand for? (delta time?)
To make things more clear and easy to understand, what do you think about renaming to something like `minutes_until_expire` / `time_until_expire_minutes` / `time_remaining_minutes` etc.

File: src/ops/terraform/terraform_cmd_generator.py (hunk 1 above)
Comment (costimuraru) [reply src/ops/terraform/terraform_cmd_generator.py]: Also, for terraform, I don't think we're using `AWS_PROFILE`. terraform fetches this data from the `provider.tf` file, which is generated via `himl`

File: src/ops/cli/aws.py (hunk 2 above)
Comment (mujx) [reply src/ops/cli/aws.py]: I assume `dttm` means datetime, I used the same terminology as in the AWS config. I'll rename to `minutes_until_expire`

File: src/ops/terraform/terraform_cmd_generator.py (hunk 1 above)
Comment (mujx) [reply src/ops/terraform/terraform_cmd_generator.py]: What should I use to differentiate between AWS and other cloud providers?

**Comment by mujx** [comment]
@amuraru Do you have a way in mind to move forward with this? A more robust way to detect if we're running on AWS.

File: src/ops/terraform/terraform_cmd_generator.py (hunk 1 above)
Comment (amuraru) [reply src/ops/terraform/terraform_cmd_generator.py]: @serban21 can you chime in pls?

**Pull request closed without merging**